packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -814,6 +853,17 @@ export default util.createRule<Options, MessageId>({
         }
       }
 
+      const enums = types.filter(type =>
+        tsutils.isTypeFlagSet(
+          type,
+          ts.TypeFlags.Enum | ts.TypeFlags.EnumLike | ts.TypeFlags.EnumLiteral,
```

Comment:
I tried removing the other type flags locally and all tests still passed. Either they can be removed safely or there's some missing cases to be tested.

```suggestion
          ts.TypeFlags.EnumLike,
```